iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/rules/ThrottleResistantTestRule.java

```diff
@@ -49,6 +50,19 @@ public void evaluate() throws Throwable {
                     log.warn("Thottling detected in test {}, waiting for {} milliseconds and then re-running the test", description.getMethodName(), THROTTLING_RETRY_DELAY_MILLISECONDS, e);
                     Thread.sleep(THROTTLING_RETRY_DELAY_MILLISECONDS);
                 }
+                catch (Exception e)
+                {
+                    if (e.getMessage().toLowerCase().contains(CONNECTION_REFUSED))
```

Comment:
We don't have a way to distinguish exceptions except by the message?

Reply:
This is for the service client (AMQP) in particular, and we don't have any sort of proper exception hierarchy there, no. This is a pretty special case, I'll admit. I don't have a better way to accomplish this though